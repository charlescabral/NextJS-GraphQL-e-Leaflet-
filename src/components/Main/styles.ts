import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #f4f4f4;
  color: #000;
  width: 100%;
  height: 100%;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 4rem;
`

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 4rem;
`

export const Subtitle = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`

export const Illustration = styled.img`
  margin-top: 2rem;
  width: min(30rem, 100%);
`
