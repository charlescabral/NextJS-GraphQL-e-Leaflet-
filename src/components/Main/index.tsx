import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src="/vercel.svg" alt="Vercel" />
    <S.Title>React Boilerplate</S.Title>
    <S.Subtitle>A complete boilerplate</S.Subtitle>
    <S.Illustration src="AwetiMask.svg" alt="Aweti Mask" />
  </S.Wrapper>
)

export default Main
