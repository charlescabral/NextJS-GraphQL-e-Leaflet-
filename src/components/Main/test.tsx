import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('shold render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /charles/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
