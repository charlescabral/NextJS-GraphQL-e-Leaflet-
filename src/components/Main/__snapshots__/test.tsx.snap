// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> shold render the heading 1`] = `
<main
  class="styles__Wrapper-dzx1ao-0 daNyOe"
>
  <img
    alt="Vercel"
    class="styles__Logo-dzx1ao-1 dUetyB"
    src="/vercel.svg"
  />
  <h1
    class="styles__Title-dzx1ao-2 fXIGei"
  >
    React Boilerplate
  </h1>
  <h2
    class="styles__Subtitle-dzx1ao-3 eojGG"
  >
    A complete boilerplate
  </h2>
  <img
    alt="Aweti Mask"
    class="styles__Illustration-dzx1ao-4 fCSVlw"
    src="AwetiMask.svg"
  />
</main>
`
